ddl/partition.go

```diff
@@ -1476,6 +1476,13 @@ func checkAddPartitionTooManyPartitions(piDefs uint64) error {
 	return nil
 }
 
+func checkAddPartitionOnTemporaryMode(tbInfo *model.TableInfo) error {
+	if tbInfo.Partition != nil && tbInfo.TempTableType != model.TempTableNone {
+		return errors.Trace(ErrPartitionNoTemporary)
```

Comment:
It seems that the usage is incorrect. Please add a variable in ddl/error.go. You can refer to other usages.